spec/services/service_offering_spec.rb

```diff
@@ -0,0 +1,44 @@
+describe ServiceOffering do
+  let(:api_instance) { double(:api_instance, :show_service_offering => service_offering) }
+  let(:service_offering) { ServiceOffering.new(params) }
+  let(:params) { {'service_offering_ref' => '10'} }
+
```

Comment:
I would suggest using `describe` as the level to show the method you're testing, and then the `it` block would have a brief description of what is being tested. Example:
```ruby
describe "#show" do
  it "calls api_instance.show_service_offering with the correct value" do
    # ...
  end
end
```

This allows us to have the setup steps for each method (or further `describe` contexts) more clearly defined instead of having to hunt in the `it` blocks for what is being set up. Granted, there's only one spec for each method right now, but later down the line I could see this being pretty messy.

Reply:
I usually include a describe block if I am testing multiple characteristics of a method, the describe declares the the method name and then the `it` blocks log each named attempt against that describe block.  In these cases I am only testing a single operation on the specific method, whereas I believe the name of the method is description enough.  I tried adding the describe block, but it felt to me like it was overcomplicating the specs I was running.

Reply:
I guess I disagree here, especially about the name of the method being enough of a description, especially since the code is relying on an outside `api_instance` that this class doesn't really care about or know if the calling method is complex or not. `#show` doesn't simply just look up the object in the database which would probably be enough to leave it as the `it` description. It is calling out to this `api_instance` which I would argue is important to document via that `it` description.